### PR TITLE
Fixed an issue where the socket address family was being corrupted due to padding fields being incorrect used.

### PR DESCRIPTION
**Pull request by john-sharratt**

### CODE REVIEW

File: lib/wasi-types/src/types.rs [hunk 1]
```diff
@@ -233,6 +233,7 @@ pub mod net {
     #[repr(C)]
     pub struct __wasi_addr_t {
         pub tag: Addressfamily,
+        pub _padding: u8,
```
Comment (theduke) [inline lib/wasi-types/src/types.rs:236]: Please add a comment to the `_padding` field explaining why it is there.

File: lib/wasi-types/src/types.rs [hunk 2]
```diff
@@ -246,6 +247,7 @@ pub mod net {
     #[repr(C)]
     pub struct __wasi_addr_port_t {
         pub tag: Addressfamily,
+        pub _padding: u8,
```
Comment (theduke) [inline lib/wasi-types/src/types.rs:250]: Please add a comment to the `_padding` field explaining why it is there.

File: lib/wasi-types/src/types.rs [hunk 3]
```diff
@@ -259,6 +261,7 @@ pub mod net {
     #[repr(C)]
     pub struct __wasi_cidr_t {
         pub tag: Addressfamily,
+        pub _padding: u8,
```
Comment (theduke) [inline lib/wasi-types/src/types.rs:264]: Please add a comment to the `_padding` field explaining why it is there.

**Review by theduke — CHANGES_REQUESTED** [review]

File: lib/wasi-types/src/types.rs (hunk 1 above)
Comment (Michael-F-Bryan) [reply lib/wasi-types/src/types.rs:236]: @theduke if we are manually adding padding bytes, should we mark the struct as `#[repr(packed)]` and have an `assert_eq!(std::mem::size_of::<__wasi_addr_t>(), 42)` test?

**Review by theduke — APPROVED** [review]

**Review by ptitSeb — APPROVED** [review]

**Comment by syrusakbary** [comment]
It seems this is already in master. Closing

**Pull request closed without merging**